src/main/java/com/google/devtools/build/lib/worker/WorkerSpawnRunner.java

```diff
@@ -530,7 +532,12 @@ private WorkResponse executeRequest(
                 String.format("Worker #%d preparing execution", worker.getWorkerId()))) {
       // We consider `prepareExecution` to be also part of setup.
       Stopwatch prepareExecutionStopwatch = Stopwatch.createStarted();
-      worker.prepareExecution(inputFiles, outputs, key.getWorkerFilesWithDigests().keySet());
+      Set<PathFragment> workerFiles = key.getWorkerFilesWithDigests().keySet();
```

Comment:
I am a little confused; shouldn't the path mapping transformation be applied *before* computing the worker key, not after? In other words, if two actions have the same set of tool inputs after path mapping (but not before), shouldn't they share the same set of persistent workers?

This seems desirable because currently `--worker_max_instances=Foo=N` doesn't actually mean "at most N workers will be spawned", but rather "at most N*M where M is the number of distinct configurations", and normalizing the key would at least mitigate it. (But maybe there's a subtle reason we can't do this and maintain correctness?)

It would also remove the asymmetry between tool inputs and other inputs in this file (it looks like `inputFiles` have already been path-mapped elsewhere?) which would help readability.

Reply:
Thanks for the suggestion, looks much better now.